src/Novu/Interfaces/ISubscriberClient.cs

```diff
@@ -1,66 +1,91 @@
 ﻿using Novu.DTO;
+using Novu.DTO.Notifications;
+using Novu.DTO.Subscriber;
+using Novu.DTO.Subscriber.Preferences;
 using Novu.Exceptions;
 using Refit;
 
-namespace Novu.Clients;
+namespace Novu.Interfaces;
 
 public interface ISubscriberClient
 {
     /// <summary>
-    /// Get a paginated response of current Subscribers
+    ///     Get a paginated response of current Subscribers
     /// </summary>
     /// <returns></returns>
     /// <exception cref="HttpRequestException">
-    ///  Thrown when the status code does not equal 200.
+    ///     Thrown when the status code does not equal 200.
     /// </exception>
     /// <exception cref="NovuClientException"></exception>
     [Get("/subscribers")]
-    Task<SubscribersDto> GetSubscribers([Query] int page = 0);
+    Task<PaginatedResponseDto<SubscriberDto>> GetSubscribers([Query] int page = 0, int limit = 100);
 
     /// <summary>
-    /// Get a single Subscriber
+    ///     Get a single Subscriber
     /// </summary>
-    /// <param name="id"><see cref="String"/> Subscriber ID</param>
+    /// <param name="id"><see cref="String" /> Subscriber ID</param>
     /// <returns></returns>
     [Get("/subscribers/{id}")]
     Task<SubscriberDto> GetSubscriber(string id);
```

Comment:
Returning a single subscriber is now wrapped in a `data` prop, so this is broken atm 

curl -X GET \
 -H "Authorization: ApiKey <api_key>" \
https://api.novu.co/v1/subscribers/xyz | jq

```js
{
  "data": { <== this has changed
     "subscriberId": "xyz"
    ...
    ...
  }
}
```

Reply:
@joaonlforte I'm struggling to reproduce the error in the tests on the [fork](https://github.com/picmi/novu-dotnet)

I have 0.17.1 running (this [docker script](https://github.com/novuhq/novu/blob/53d2acde2180dc3c8b5da1a620b0d5c3fd54a306/docker/local/deployment/docker-compose.yml#L26))

[This test](https://github.com/picmi/novu-dotnet/blob/2b29f957b91d5a64089475850a2697b3116fc9f6/src/Novu.Tests/IntegrationTests/SubscriberTests.cs#L30) passes

I know that the original pull request was against 0.16 because that was the latest at that point. I then had to upgrade I think about 2-3 days later to get some of the other commits to pass. 

Personally, the test project needs to know what version it is running against.

Thanks in advance to help me work out what is wrong (or what I am doing wrong).

Reply:
Hi @toddb 👋 

I can't tell exactly the version, since I'm using the SaaS version of Novu (cloud), and i'm using [this](https://www.nuget.org/packages/Novu) package v0.2.2

The error i'm getting upon deserialising the response is 

```
Refit.ApiException: An error occured deserializing the response.
 ---> Newtonsoft.Json.JsonSerializationException: Cannot deserialize the current JSON object (e.g. {"name":"value"}) into type 'System.Collections.Generic.List`1[Novu.DTO.AdditionalDataDto]' because the type requires a JSON array (e.g. [1,2,3]) to deserialize correctly.
To fix this error either change the JSON to a JSON array (e.g. [1,2,3]) or change the deserialized type so that it is a normal .NET type (e.g. not a primitive type like integer, not a collection type like an array or List<T>) that can be deserialized from a JSON object. JsonObjectAttribute can also be added to the type to force it to deserialize from a JSON object.
Path 'data._id', line 1, position 15.
```

but the part i find odd is this one `Path 'data._id', line 1, position 15.`. It seems that the entire response of the `GetSubscriber` is now wrapped in a `data` property (similarly what is happening with the `CreateTopic` endpoint where a `TopicCreateResponseDto` is returned )

<img width="400" alt="Screenshot 2023-08-22 at 10 35 02" src="https://github.com/novuhq/novu-dotnet/assets/5887317/62d3a370-8bf4-4416-bbe0-4e4ca99f853e">

In summary seems that the response contract has changed for the Subscriber endpoints and now the actual response object is wrapped in a `Data` prop

<img width="586" alt="Screenshot 2023-08-22 at 10 38 31" src="https://github.com/novuhq/novu-dotnet/assets/5887317/62b5665b-e011-4d65-9fb2-ca1b2d093142">
<img width="555" alt="Screenshot 2023-08-22 at 10 38 50" src="https://github.com/novuhq/novu-dotnet/assets/5887317/e70b0786-1b96-443c-b68b-b5cfb5528488">

So instead of returning a `SubscriberDto` should be returning it in an envelope, something like

```csharp 
class Foo {
    [JsonProperty("data")]
    public SubscriberDto Data { get; set; }
}
```
```csharp

    [Get("/subscribers/{id}")]
    Task<Foo> GetSubscriber(string id);

    [Post("/subscribers")]
    Task<Foo> CreateSubscriber([Body] CreateSubscriberDto dto);

```

Reply:
I really suspect you are running against 0.16.x and the client code I am writing against is 0.17.x. I can imagine the frustration. I have had a quick look and I can see no versioning of the API via the API responses. 

You have introduced the class `Foo` but what you are looking for is `NovuResponse<T>` or `PaginatedResponseDto<T>`. However, where confusion lies IMHO is that in some cases neuther are used and the `Data<T>` is inline (eg `TriggerBulkResponseDto` or `SubscribersDto`). 

I hope there is some help here.

ps. when I had this problem, I had to get out the source to check what as going on. Sorry.

Reply:
@joaonlforte So, I have found where the API is versioned. It is on the `health-check` resource. Looking at the source, it is not an external API resource (so you can't use the ApiKey token) and need to hijack your JWT. I did that by copying a request and then changing it. I don't get why this isn't at the root of the API.

```
  ~ curl 'http://localhost:3000/v1/health-check' \
  -H 'Accept: application/json, text/plain, */*' \
   -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI2NGM5YTc1ZmUxOGQ3ZjJjYjk3ODcxNjMiLCJmaXJzdE5hbWUiOiJ0IiwibGFzdE5hbWUiOiJub3Z1IiwiZW1haWwiOiJ0K25vdnUtbG9jYWwuZGV2QHBpY21pLmlvIiwib3JnYW5pemF0aW9uSWQiOiI2NGM5YTc3NWUxOGQ3ZjJjYjk3ODcxNzAiLCJyb2xlcyI6WyJhZG1pbiJdLCJlbnZpcm9ubWVudElkIjoiNjRjOWE3NzZlMThkN2YyY2I5Nzg3MTc2IiwiaWF0IjoxNjkyNTc0NjU1LCJleHAiOjE2OTUxNjY2NTUsImlzcyI6Im5vdnVfYXBpIn0.9-iVXXsWKfR7DaBojrhlMtRjMBjXSAJGiZXt0GIxZYU' \
  --compressed
{"status":"error","info":{"db":{"status":"up"},"apiVersion":{"version":"0.17.1","status":"up"},"triggerQueue":{"status":"up"}},"error":{"inMemory":{"status":"down"}},"details":{"db":{"status":"up"},"apiVersion":{"version":"0.17.1","status":"up"},"triggerQueue":{"status":"up"},"inMemory":{"status":"down"}}}%  
```

Reply:
So the answer in production is: 0.18.0

```
~ curl 'https://api.novu.co/v1/health-check' \
  -H 'authority: api.novu.co' \
  -H 'accept: application/json, text/plain, */*' \
  -H 'accept-language: en-GB,en-US;q=0.9,en;q=0.8,mi;q=0.7' \
  -H 'authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6I ...-bfSmesKN2w' \
   --compressed
   
{"data":{"status":"ok","info":{"db":{"status":"up"},"inMemory":{"status":"up"},"apiVersion":{"version":"0.18.0","status":"up"},"triggerQueue":{"status":"up"},"cacheService":{"status":"up"}},"error":{},"details":{"db":{"status":"up"},"inMemory":{"status":"up"},"apiVersion":{"version":"0.18.0","status":"up"},"triggerQueue":{"status":"up"},"cacheService":{"status":"up"}}}}%                 ➜  ~ 
```

Reply:
> I really suspect you are running against 0.16.x and the client code I am writing against is 0.17.x. I can imagine the frustration. I have had a quick look and I can see no versioning of the API via the API responses.
> 
> You have introduced the class `Foo` but what you are looking for is `NovuResponse<T>` or `PaginatedResponseDto<T>`. However, where confusion lies IMHO is that in some cases neuther are used and the `Data<T>` is inline (eg `TriggerBulkResponseDto` or `SubscribersDto`).
> 
> I hope there is some help here.
> 
> ps. when I had this problem, I had to get out the source to check what as going on. Sorry.

Thanks @toddb 

that might be the issue but i don't own the client, so i can't change the contract, I was hoping a fix would be released. Aren't you the maintainer of the package?

Thanks in advance ;)

Reply:
Nope. I'm not. I've just made a pull request so that it works for me! But you are right that it is broken for 0.18. I will put through a fix in the next 24 hours (after I have a sleep).

Reply:
@joaonlforte Thanks—I upgraded to 0.18 (which coincidentally is not updated in the docker-compose[ deployment script](https://github.com/novuhq/novu/blob/next/docker/local/deployment/docker-compose.yml)). 

This library is broken against 0.18.0.

@wh1337 It seems that this library needs to be able create the test environment locally. That means maintaining copy of the docker-compose doesn't it?